File: transfusion-center/src/main/java/isa/transfusioncenter/Dto/RegisterUserDto.java
package isa.transfusioncenter.Dto;

import lombok.Data;

@Data
public class RegisterUserDto {
    private String email;
    private String password;
    private String name;
    private String lastname;
    private String address;
    private String city;
    private String country;
    private String phone;
    private String socialSecurityNumber;
    private String gender;
    private String occupation;
    private String companyInfo;
}

File: bachelors-project/src/modules/users/dto/register-user.dto.ts
import { Field, InputType } from "@nestjs/graphql";
import { Gender, RegisterUser } from "src/modules/utils";


@InputType()
export class RegisterUserDto implements RegisterUser{

    @Field()
    email: string;

    @Field()
    password: string;

    @Field()
    verifyPassword: string;

    @Field()
    firstName: string;

    @Field()
    lastName: string;

    @Field()
    phoneNumber: string;

    @Field()
    address: string;

    @Field()
    city: string;

    @Field()
    countryCode: string;

    @Field()
    socialSecurityNumber: string;

    @Field()
    occupation: string;

    @Field()
    companyInfo: string;

    @Field(()=>Gender)
    gender: Gender;

}